pdns/dnsdist-tcp.cc

```diff
@@ -750,6 +754,13 @@ static void handleQuery(std::shared_ptr<IncomingTCPConnectionState>& state, cons
     TCPResponse response;
     response.d_selfGenerated = true;
     response.d_buffer = std::move(state->d_buffer);
+    setIDStateFromDNSQuestion(response.d_idstate, dq, std::move(qname));
+    response.d_idstate.origID = dh->id;
```

Comment:
The pointer might have been invalidated by the move above, so we should get the ID before the moves.

Reply:
I'm afraid this is still a bit too late if `dh` is pointing to the data that was in `state->d_buffer`?